src/contracts/pods/EigenPodManager.sol

```diff
@@ -654,26 +640,22 @@ contract EigenPodManager is
      * @notice Internal function to enter `podOwner` into undelegation limbo
      * @dev Does nothing if the `podOwner` has no delegated shares (i.e. they are already in undelegation limbo or have no shares)
      * OR if they are not actively delegated to any operator.
+     *
+     * This method assumes the podOwner is delegated, as it's being called by the
+     * DelegationManager (and supplied with `delegatedTo`)
      */
-    function _enterUndelegationLimbo(address podOwner) internal {
-        if (!podOwnerHasNoDelegatedShares(podOwner) && delegationManager.isDelegated(podOwner)) {
-            // look up the address that the pod owner is currrently delegated to in EigenLayer
-            address delegatedAddress = delegationManager.delegatedTo(podOwner);
-
+    function _enterUndelegationLimbo(address podOwner, address delegatedTo) internal returns (uint) {
+        if (podOwnerShares[podOwner] != 0 && !isInUndelegationLimbo(podOwner)) {
             // store the undelegation limbo details
             _podOwnerUndelegationLimboStatus[podOwner].active = true;
             _podOwnerUndelegationLimboStatus[podOwner].startBlock = uint32(block.number);
-            _podOwnerUndelegationLimboStatus[podOwner].delegatedAddress = delegatedAddress;
+            // TODO: is this even necessary to keep in state? i think it's literally only used to emit an event later
+            _podOwnerUndelegationLimboStatus[podOwner].delegatedAddress = delegatedTo;
```

Comment:
This is necessary because we treat withdrawals as completeable or not (basically, "can they be slashed") based on the address that the staker was delegated to _at the time the withdrawal was created_. "Undelegation Limbo" is subject to the same delays/conditions as a withdrawal.

Reply:
that makes sense. i definitely did the EPM first and when looking at the SM, started wondering why it doesn't have a more concrete concept of "undelegation limbo." Or, why the EPM's undelegation limbo didn't look more like the SM's queued withdrawal concept.

As I said, I think there's further room for simplifying this - but we can pin it for post M2!